drivers/usb/udc/udc_mcux_ehci.c

```diff
@@ -92,6 +97,50 @@ static int udc_mcux_control(const struct device *dev, usb_device_control_type_t
 	return 0;
 }
 
+/* Helper function to keep track of the activity state for the udc */
+static void udc_mcux_change_state(const struct device *dev, bool enabled, bool vbus_present)
+{
+	struct udc_mcux_data *data = udc_get_private(dev);
+
+	if (data->enabled == enabled && data->vbus_present == vbus_present) {
+		return;
+	}
+
+	if (vbus_present != data->vbus_present && enabled) {
```

Comment:
That is wrong, must not depend on enabled, it is usually other way around.

Reply:
I am unclear at the error but I have updated the conditional to be only based on the vbus_present variable.